toolkit-test-app/src/main/java/com/esri/arcgisruntime/toolkit/test/ar/ArcGISArViewActivity.kt

```diff
@@ -0,0 +1,353 @@
+                        arcGisArView.translationTransformationFactor = 2000.0
+                    }
+                }
+
+                arcGisArView.locationDataSource = null
+            }
+        }
+    }
+
+    /**
+     * AR Mode: Tabletop AR
+     * Scene that is centered on Yosemite National Park.
+     *
+     * @since 100.6.0
+     */
+    private fun yosemiteScene(): () -> ArcGISScene {
+        return {
+            ArcGISScene().apply {
+                addElevationSource(this)
+
+                val layer =
+                    IntegratedMeshLayer("https://tiles.arcgis.com/tiles/FQD0rKU8X5sAQfh8/arcgis/rest/services/VRICON_Yosemite_Sample_Integrated_Mesh_scene_layer/SceneServer")
+                this.operationalLayers.add(layer)
+                layer.addDoneLoadingListener {
+                    layer.loadError?.let {
+                        it.message?.let { errorMessage ->
+                            displayErrorMessage(errorMessage)
+                            return@addDoneLoadingListener
+                        }
+                    }
+
+                    layer.fullExtent?.center?.let { center ->
+                        with(baseSurface.elevationSources.first()) {
+                            addDoneLoadingListener {
+                                loadError?.let {
+                                    it.message?.let { errorMessage ->
+                                        displayErrorMessage(errorMessage)
+                                    }
+                                }
+
+                                with(this@apply.baseSurface.getElevationAsync(center)) {
+                                    // when the elevation has loaded
+                                    this.addDoneListener {
+                                        loadError?.let {
+                                            it.message?.let { errorMessage ->
+                                                displayErrorMessage(errorMessage)
+                                            }
+                                        }
+
+                                        this.get()?.let { elevation ->
+                                            val camera = Camera(
+                                                center.y,
+                                                center.x,
+                                                elevation,
+                                                0.0,
+                                                0.0,
+                                                0.0
+                                            )
+                                            arcGisArView.originCamera = camera
+                                            arcGisArView.translationTransformationFactor = 1000.0
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
```

Comment:
There is a lot of nesting going on here, can we avoid this by using less standard Kotlin methods? I got that idea from this post you shared with me... :)
https://blog.danlew.net/2019/02/12/when-to-use-kotlins-stdlib-functions/